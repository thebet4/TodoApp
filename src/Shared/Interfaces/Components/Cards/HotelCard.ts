export default interface IHotelCardProps {
  category?: number;
  name: string;
  description: string;
  valueInInn: string;
  valueInBrl: string;
  icon: any;
  location?: {
    lat: number;
    lng: number;
  },
  imageUrl: any;
}
