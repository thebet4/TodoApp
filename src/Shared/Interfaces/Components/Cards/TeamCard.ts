export default interface ITeamCardProps {
  name: string;
  role: string;
  imageUrl: any;
}
