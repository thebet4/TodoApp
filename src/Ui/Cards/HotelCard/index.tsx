import React from 'react';
import { Image } from 'react-native';
import IHotelCardProps from '../../../Shared/Interfaces/Components/Cards/HotelCard';
import { Colors } from '../../Theme';
import Body16 from '../../Typography/Body16';
import Subtitle from '../../Typography/Subtitle';
import { CardContainer, CardContentContainer } from './style';

const HotelCard = ({
  imageUrl, description, name, valueInBrl, valueInInn,
}: IHotelCardProps) => {
  return (
    <CardContainer>
      <Image
        source={imageUrl || require('./Assets/HotelImage.png')}
        style={{ width: '100%', maxHeight: 240 }}
        resizeMode="cover"
      />
      <CardContentContainer>
        <Subtitle style={{ color: Colors.Base.Gray }}>
          {name}
          {' '}
        </Subtitle>
        <Body16 style={{ color: Colors.Primary.Normal, marginVertical: 12 }}>
          {description}
        </Body16>
        <Subtitle style={{ color: Colors.Base.Gray }}>
          INN
          {' '}
          {valueInInn}
        </Subtitle>
        <Body16 style={{ color: Colors.Base.Gray }}>
          R$
          {' '}
          {valueInBrl}
        </Body16>
      </CardContentContainer>
    </CardContainer>
  );
};

export default HotelCard;
