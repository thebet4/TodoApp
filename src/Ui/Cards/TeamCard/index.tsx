import React from 'react';
import { Image, View } from 'react-native';
import ITeamCardProps from '../../../Shared/Interfaces/Components/Cards/TeamCard';
import { Colors } from '../../Theme';
import Body16 from '../../Typography/Body16';
import Subtitle from '../../Typography/Subtitle';

const TeamCard = ({ name, imageUrl, role } : ITeamCardProps) => {
  return (
    <View>
      <Image
        style={{
          marginLeft: 14,
          marginRight: 14,

          width: 312,
          height: 354,
        }}
        source={imageUrl}
      />
      <Subtitle style={{ color: Colors.Primary.Normal, textAlign: 'center' }}>
        {name}
      </Subtitle>
      <Body16 style={{ color: Colors.Base.Gray, textAlign: 'center' }}>
        {role}
      </Body16>
    </View>
  );
};

export default TeamCard;
