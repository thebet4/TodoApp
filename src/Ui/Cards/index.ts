export { default as HotelCard } from './HotelCard';
export { default as TeamCard } from './TeamCard';
