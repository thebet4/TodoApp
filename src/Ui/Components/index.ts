export { default as Header } from './Header'
export { default as TabBarItem } from './TabBarItem'
