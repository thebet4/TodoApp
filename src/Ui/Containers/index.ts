export { default as Container } from './Container';
export { ContentContainer } from './ContentContainer';
