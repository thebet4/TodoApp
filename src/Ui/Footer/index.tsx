import React from 'react';
import {
  ImageBackground, View, TouchableOpacity, Image,
} from 'react-native';
import { Colors } from '../Theme';
import Caption from '../Typography/Caption';
import Title40 from '../Typography/Title40';

const Footer = () => {
  return (

    <ImageBackground
      source={require('./Assets/HomeFooterBg.png')}
      style={{ paddingBottom: 64 }}
    >
      <View style={{
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: 92,
      }}
      >
        <View style={{
          width: '30%',
          backgroundColor: Colors.Base.Light,
          height: 1,
        }}
        />
        <Title40 style={{
          color: Colors.Base.Light,
          marginHorizontal: 12,
        }}
        >
          Partners
        </Title40>
        <View style={{
          width: '30%',
          backgroundColor: Colors.Base.Light,
          height: 1,
        }}
        />
      </View>

      <View style={{
        alignItems: 'center',
        justifyContent: 'space-around',
      }}
      >
        <Image source={require('./Assets/partner1.png')} resizeMode="center" />
        <Image source={require('./Assets/partner2.png')} resizeMode="center" />
        <Image source={require('./Assets/partner4.png')} resizeMode="center" />
        <Image source={require('./Assets/partner5.png')} resizeMode="center" />
        <Image source={require('./Assets/partner6.png')} resizeMode="center" />
        <Image source={require('./Assets/partner7.png')} resizeMode="center" />

      </View>

      <View style={{ width: '70%', alignSelf: 'center' }}>

        <TouchableOpacity style={{
          height: 44,
          alignItems: 'center',
          justifyContent: 'center',
          borderWidth: 2,
          borderColor: Colors.Base.Light,
          borderRadius: 51,
          marginBottom: 17,
        }}
        >
          <Caption style={{ color: Colors.Base.Light, fontSize: 18 }}>
            Acessar
          </Caption>
        </TouchableOpacity>
        <TouchableOpacity style={{
          height: 44,
          alignItems: 'center',
          justifyContent: 'center',
          borderRadius: 51,
          backgroundColor: Colors.Base.Light,
        }}
        >
          <Caption style={{ color: Colors.Primary.Normal, fontSize: 18 }}>
            Abrir Conta
          </Caption>
        </TouchableOpacity>
      </View>
    </ImageBackground>

  );
};

export default Footer;
