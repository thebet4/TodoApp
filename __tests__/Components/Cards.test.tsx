import renderer from 'react-test-renderer';
import React from 'react';

import { HotelCard, TeamCard } from '../../src/Ui/Cards';

describe('<HotelCard />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HotelCard imageUrl="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<TeamCard />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TeamCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
