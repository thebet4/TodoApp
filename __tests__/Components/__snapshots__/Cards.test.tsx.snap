// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HotelCard /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EFEFEF",
        "borderBottomLeftRadius": 70,
        "borderBottomRightRadius": 70,
        "height": 500,
        "marginBottom": 71,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 71,
        "width": 600,
      },
    ]
  }
>
  <Image
    resizeMode="cover"
    source="test"
    style={
      Object {
        "maxHeight": 240,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingLeft": 36,
          "paddingTop": 36,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter_600SemiBold",
            "fontSize": 24,
            "lineHeight": 36,
          },
          Object {
            "color": "#585858",
          },
        ]
      }
    >
       
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter_400Regular",
            "fontSize": 16,
            "lineHeight": 24,
          },
          Object {
            "color": "#6A0DFB",
            "marginVertical": 12,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter_600SemiBold",
            "fontSize": 24,
            "lineHeight": 36,
          },
          Object {
            "color": "#585858",
          },
        ]
      }
    >
      INN
       
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter_400Regular",
            "fontSize": 16,
            "lineHeight": 24,
          },
          Object {
            "color": "#585858",
          },
        ]
      }
    >
      R$
       
    </Text>
  </View>
</View>
`;

exports[`<TeamCard /> renders correctly 1`] = `
<View>
  <Image
    style={
      Object {
        "height": 354,
        "marginLeft": 14,
        "marginRight": 14,
        "width": 312,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 24,
          "lineHeight": 36,
        },
        Object {
          "color": "#6A0DFB",
          "textAlign": "center",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Inter_400Regular",
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#585858",
          "textAlign": "center",
        },
      ]
    }
  />
</View>
`;
